### ISSUE TITLE
No Mouse or Keyboard Input Detection At All

### ISSUE THREAD

**Issue by wavy7s** [UniversalSplitScreen/UniversalSplitScreen#66]
Hey there.

So Universal Split Screen was working completely fine for me before I did a clean install of Windows. After the clean install, it no longer works.

No mouse or keyboard input registry at all. Don't think it is a settings issue on Universal Split Screen, rather it's a Windows setting? Not really sure.

I use it to split screen Minecraft 1.16.4, and it was working beautifully before. Now after pressing "Start split screen" with the Minecraft 1.14.3 preset, no keyboard or mouse regsitry works. Occasionally right on startup, the fake drawn mouse paints on the screen in one or two positions. Even got the mouse to work one time (very slow) and it was constantly painting it's path as I moved around the Minecraft menu. 

Note: Not even the "End" key works to end Split Screen's hold over the two MultiMC instances. I have to CTRL ALT DEL and go back to the Desktop to press 'End' right before it will stop registering keyboard input again.

I'm not really sure what to do here. Any advice?

**Comment by wavy7s** [comment]
Actually, I think I have the same problem as this guy here: https://github.com/UniversalSplitScreen/UniversalSplitScreen/issues/6

After CTRL ALT DEL mouse and keyboard input work, but very glitchy. Already tried messing with my mouse polling rate, did not help.

On my left window, both mouse and keyboard input work, except it is stuttering frequently. Set VSync on, no effect. Still stuttering just as much, if not more. 

On my right window, only the keyboard input works, mouse does not register. Both windows are sharing the same keyboard (for testing).

Also tried two different Chrome windows instead of Minecraft. No change. Same issues. Also tried two different keyboards and mouse, to see if it was an issue with sharing a keyboard, that wasn't it either.

Screenshot of the mouse drawing all over the screen:
![Screenshot_1](https://user-images.githubusercontent.com/79673116/109236656-ad85ed80-7795-11eb-8294-30398ff512e7.png)

Options used (Minecraft 1.14.3 and never changed these settings):
![Screenshot_2](https://user-images.githubusercontent.com/79673116/109236976-559bb680-7796-11eb-92d1-bbbc8f178dca.png)

**Comment by scorpeti** [comment]
Hello, I have a similar issue as @wavy7s 

TL;DR: When I click on "Start split screen", the targeted window(s) do(es) not get into "active" state from Universal Split Screen's point of view, so no mouse/keyboard input works, unless I hit Ctrl+Alt+Del, then Esc. After doing that, the input doesn't fully work as expected.


Universal Split Screen was working completely fine a few months ago using a 4+ years old version of Windows 10 Pro (it did not update itself automatically, and it did not include the Creators Update). I used it for several games, including Minecraft 1.16.3.

Then I re-installed Windows 10 Pro a month ago on the same hardware, currently has 19042.928 build number. Since then whenever I click on the "Start split screen" button of Universal Split Screen, it doesn't auto-focus the previously set windows and doesn't react to any mouse or keyboard input at all. I have to press Ctrl+Alt+Del then Esc, after that the simulated mouse+keyboard mostly works on the hooked windows, but not exactly as expected: the cursor is drawn regardless the "Hook mouse visibility" option and the cursor movement is not bound to the FPS camera (I can see the virtual cursor moving, but the game doesn't react like it did on the older version of Windows, even if the Universal Split Screen options were set exactly the same both time).

I tried the following combination of applications and got exactly the same results all the time:
Universal Split Screen:
- v1.2.0
- v1.1.1

Application tried with Universal Split Screen:
- Minecraft 1.16.5 with Optifine
- Minecraft 1.16.5
- Minecraft 1.16.4
- Minecraft 1.16.3
- Google Chrome
- Windows Explorer

I also tried almost all combinations of Universal Split Screen options provided by its UI, but couldn't get any better than using the officially suggested Minecraft 1.14.3 profile and turning off the "Raw input" in Minecraft's options, where the mouse behaved as I mentioned before. However, I didn't see any visual anomalies, only the appearance of cursor and the movement of the mouse was broken for me.

If that helps in fixing this issue, I am happy to provide more details about my system/setup or to do some further tests, because when Universal Split Screen works, it does an excellent job and is a super awesome tool!

**Comment by scorpeti** [comment]
TL;DR: Workaround: download source code, build Universal Split Screen using Visual Studio, copy the missing exe files from v1.2.0 and you might get a working version.


UPDATE: I did a workaround, with the following steps I was able to recreate a "working version" of the app+system composition that now behaves the same way as it did on my old Windows. I do not completely know the reasons what is the difference between the downloaded v1.2.0, but here you go:

I realized that when I start Universal Split Screen and instantly select a window, set my mouse ID and start split screen really fast (in a matter of seconds), the application hooks to that window successfully as it used to be, without the need of CTRL + ALT + DEL. But, after a few seconds, that window gets "unfocused" somehow and loses the ability to get any input from Universal Split Screen, so in the end I needed to press CTRL+ALT+DEL anyway. This behavior was easy to reproduce, so I decided to download the source code and debug the reason.

(Also, one note that the downloaded v1.2.0 Universal Split Screen had to have been executed with administration rights on my current system in order to not get a Windows Fault error.)

I cloned the repository, rebuilt the solution of Universal Split Screen, copied the missing exe files (IJx**.exe) to the respective UniversalSplitScreen\bin\Debug folder (later I built for Release as well, copied the same files there as well) from the previously downloaded v1.2.0 release, and could start Universal Split Screen without any CTRL+ALT+DEL related issue mentioned before. My guess is that somehow above mentioned Windows' focus-losing-event did not happened this case.

One more configuration needed in order to get Minecraft 1.16.5 working: I had to turn Mouse Raw: ON and I used the Minecraft Alternate profile. Using this I get an almost stable mouse movement on both instances of the game, just as it worked months before on my old Windows.

If you would like to get more details about my setup/system, don't hesitate to ping me, sooner the better. If you would like to have some test on my environment, I am open to associate to debug this issue, because the tool is still awesome.

Cheers!

**Comment by Haikalrama** [comment]
> TL;DR: Workaround: download source code, build Universal Split Screen using Visual Studio, copy the missing exe files from v1.2.0 and you might get a working version.
> 
> UPDATE: I did a workaround, with the following steps I was able to recreate a "working version" of the app+system composition that now behaves the same way as it did on my old Windows. I do not completely know the reasons what is the difference between the downloaded v1.2.0, but here you go:
> 
> I realized that when I start Universal Split Screen and instantly select a window, set my mouse ID and start split screen really fast (in a matter of seconds), the application hooks to that window successfully as it used to be, without the need of CTRL + ALT + DEL. But, after a few seconds, that window gets "unfocused" somehow and loses the ability to get any input from Universal Split Screen, so in the end I needed to press CTRL+ALT+DEL anyway. This behavior was easy to reproduce, so I decided to download the source code and debug the reason.
> 
> (Also, one note that the downloaded v1.2.0 Universal Split Screen had to have been executed with administration rights on my current system in order to not get a Windows Fault error.)
> 
> I cloned the repository, rebuilt the solution of Universal Split Screen, copied the missing exe files (IJx**.exe) to the respective UniversalSplitScreen\bin\Debug folder (later I built for Release as well, copied the same files there as well) from the previously downloaded v1.2.0 release, and could start Universal Split Screen without any CTRL+ALT+DEL related issue mentioned before. My guess is that somehow above mentioned Windows' focus-losing-event did not happened this case.
> 
> One more configuration needed in order to get Minecraft 1.16.5 working: I had to turn Mouse Raw: ON and I used the Minecraft Alternate profile. Using this I get an almost stable mouse movement on both instances of the game, just as it worked months before on my old Windows.
> 
> If you would like to get more details about my setup/system, don't hesitate to ping me, sooner the better. If you would like to have some test on my environment, I am open to associate to debug this issue, because the tool is still awesome.
> 
> Cheers!

I know it abit to late but can i ask for your modified aplicstion of universal split screen?
